File: worker-dotnet/Services/TaskStoreService.cs
using worker_dotnet.Data;
using worker_dotnet.Domain;

namespace worker_dotnet.Services;

public class TaskStoreService
{
    private readonly ApplicationDbContext _context;

    public TaskStoreService(ApplicationDbContext context)
    {
        _context = context;
    }

    public void StartTask(WorkflowTask task)
    {
        var dbTask = _context.Tasks.Find(task.Id);
        if (dbTask != null)
        {
            dbTask.Status = worker_dotnet.Domain.TaskStatus.IN_PROGRESS.ToString();
            dbTask.UpdatedAt = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
            _context.SaveChanges();
        }
    }

    public void CompleteTask(WorkflowTask task)
    {
        var dbTask = _context.Tasks.Find(task.Id);
        if (dbTask != null)
        {
            dbTask.Status = worker_dotnet.Domain.TaskStatus.COMPLETED.ToString();
            dbTask.UpdatedAt = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
            _context.SaveChanges();
        }
    }
}

File: worker/src/main/java/com/smartcommunity/worker/services/TaskStoreService.java
package com.smartcommunity.worker.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import com.smartcommunity.worker.domain.Task;
import com.smartcommunity.worker.domain.TaskStatus;
import com.smartcommunity.worker.repository.TaskRepository;

/**
 * Service for managing tasks in the data store.
 */
@Service
@Transactional
public class TaskStoreService {

	@Autowired
	private TaskRepository taskRepository;

	@Transactional(propagation = Propagation.REQUIRES_NEW)
	public Task startTask(String id) {
		Task dbTask = taskRepository.findById(id).orElse(null);

		if (dbTask != null) {
			dbTask.setUpdatedAt(System.currentTimeMillis());
			dbTask.setStatus(TaskStatus.IN_PROGRESS.name());
			return taskRepository.saveAndFlush(dbTask);
		}
		else {
			return null;
		}
	}

	@Transactional(propagation = Propagation.REQUIRES_NEW)
	public Task completeTask(Task task) {
		Task dbTask = taskRepository.findById(task.getId()).orElse(null);

		if (dbTask != null) {
			dbTask.setUpdatedAt(System.currentTimeMillis());
			dbTask.setStatus(TaskStatus.COMPLETED.name());
			return taskRepository.saveAndFlush(dbTask);
		}
		else {
			return null;
		}
	}

}
